jmclient/jmclient/wallet_service.py

```diff
@@ -334,7 +335,8 @@ def sync_wallet(self, fast=True):
         # before startup
         self.old_txs = [x['txid'] for x in self.bci.list_transactions(100)
             if "txid" in x]
-        self.bci.post_sync_wallet_callback(self.wallet)
+        if self.bci.__class__ == BitcoinCoreNoHistoryInterface:
```

Comment:
I believe `isinstance()` is more technically correct, since it'd handle the case that `self.bci` was an instance of a child class of `BitcoinCoreNoHistoryInterface`. This is of course just theoretical for now, so I wouldn't be upset if you don't change it :)